sw/device/tests/alert_handler_lpg_sleep_mode_pings.c

```diff
@@ -0,0 +1,425 @@
+                                         kDifAlertHandlerClassB};
+  dif_alert_handler_config_t config = {
+      .alerts = alerts,
+      .alert_classes = alert_classes,
+      .alerts_len = ARRAYSIZE(alerts),
+      .local_alerts = loc_alerts,
+      .local_alert_classes = loc_alert_classes,
+      .local_alerts_len = ARRAYSIZE(loc_alerts),
+      .classes = classes,
+      .class_configs = class_configs,
+      .classes_len = ARRAYSIZE(class_configs),
+      .ping_timeout = ping_timeout,
+  };
+
+  alert_handler_testutils_configure_all(&alert_handler, config,
+                                        kDifToggleEnabled);
+  // Enables alert handler irq.
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassa, kDifToggleEnabled));
+
+  CHECK_DIF_OK(dif_alert_handler_irq_set_enabled(
+      &alert_handler, kDifAlertHandlerIrqClassb, kDifToggleEnabled));
+}
+
+void alert_handler_clear_cause_regs(void) {
+  bool is_cause;
+  // Loop over all alert_cause regs
+  for (int jj = 0; jj < ALERT_HANDLER_PARAM_N_ALERTS; jj++) {
+    CHECK_DIF_OK(
+        dif_alert_handler_alert_is_cause(&alert_handler, jj, &is_cause));
+    if (is_cause) {
+      LOG_INFO("Fired alert[%d]", jj);
+      CHECK_DIF_OK(dif_alert_handler_alert_acknowledge(&alert_handler, jj));
+      CHECK_DIF_OK(
+          dif_alert_handler_alert_is_cause(&alert_handler, jj, &is_cause));
+      LOG_INFO("After ack, is_cause = %d", is_cause);
+    }
+  }
+
+  // Loop over all loc_alert_cause regs
+  for (int jj = 0; jj < 7; jj++) {
+    // for (int jj=ALERT_HANDLER_PARAM_N_ALERTS-1; jj>0; jj--) {
+    CHECK_DIF_OK(
+        dif_alert_handler_local_alert_is_cause(&alert_handler, jj, &is_cause));
+    if (is_cause) {
+      LOG_INFO("Fired local_alert[%d]", jj);
+      CHECK_DIF_OK(
+          dif_alert_handler_local_alert_acknowledge(&alert_handler, jj));
+      CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(&alert_handler, jj,
+                                                          &is_cause));
+      LOG_INFO("After ack, local is_cause = %d", is_cause);
+    }
+  }
+}
+
+void enter_low_power(bool deep_sleep) {
+  dif_pwrmgr_domain_config_t cfg;
+  CHECK_DIF_OK(dif_pwrmgr_get_domain_config(&pwrmgr, &cfg));
+  cfg = cfg & (kDifPwrmgrDomainOptionIoClockInLowPower |
+               kDifPwrmgrDomainOptionUsbClockInLowPower |
+               kDifPwrmgrDomainOptionUsbClockInActivePower) |
+        (!deep_sleep ? kDifPwrmgrDomainOptionMainPowerInLowPower : 0);
+
+  pwrmgr_testutils_enable_low_power(&pwrmgr, kDifPwrmgrWakeupRequestSourceFive,
+                                    cfg);
+  LOG_INFO("Issue WFI to enter sleep %d", kDifPwrmgrWakeupRequestSourceFive);
+  wait_for_interrupt();
+}
+
+void check_wakeup_reason(void) {
+  dif_pwrmgr_wakeup_reason_t wakeup_reason;
+  CHECK_DIF_OK(dif_pwrmgr_wakeup_reason_get(&pwrmgr, &wakeup_reason));
+  CHECK(pwrmgr_testutils_is_wakeup_reason(
+            &pwrmgr, kDifPwrmgrWakeupRequestSourceFive) == true,
```

Comment:
Can you add a comment what `kDifPwrmgrWakeupRequestSourceFive` refers to (just for ease of parsing this).